### PR TITLE
Python 3.7 returns -1 refs than 3.6

### PR DESCRIPTION
**Pull request by ganwell**
* Python 3.7+ doesn't count the local reference, we are asking about, as
  reference. One can argue the correctness of this, we just fix it.
* Close objects if the reference count tests fail

### CODE REVIEW

File: tests/test_chirp.py [hunk 1]
```diff
@@ -27,21 +27,25 @@ def test_too_high_timeout(loop, config):
...
+    try:
+        assert a.identity != b'\0' * lib.CH_ID_SIZE
+        assert len(gc.get_referrers(a)) > 1 + ref_count_offset
+        a.stop()
+        assert len(gc.get_referrers(a)) == 1 + ref_count_offset
+        assert loop.running
+        loop.stop()
+        assert len(gc.get_referrers(loop)) == 1 + ref_count_offset
+        assert not loop.running
+    except Exception:
```
Comment (sliverc) [inline tests/test_chirp.py]: the test won't fail if there is an exception... is that what you desire? Or do you simply wanna clean up and a `finally ` would do better (or maybe even better a `pytest.fixture`?

**Comment by ganwell** [comment]
@sliverc If wonder about BaseException, it was recently discussed in the python mailinglist, that if you raise anyways you should cleanup for everything, including SystemExit and KeyboardInterrupt.

File: tests/test_loop.py [hunk 2]
```diff
@@ -8,20 +8,24 @@
...
+        assert len(gc.get_referrers(a)) > 1 + ref_count_offset
+        assert a.running
+        a.stop()
+        assert not a.running
+        assert len(gc.get_referrers(a)) == 1 + ref_count_offset
+        assert caplog.record_tuples == [
+            ('libchirp', 10, 'libuv event-loop started'),
+            ('libchirp', 10, 'libuv event-loop stopped'),
+        ]
+    except Exception:
```
Comment (sliverc) [inline tests/test_loop.py]: do you wanna use `BaseException` here as well?

Comment (ganwell) [reply tests/test_loop.py]: Woot, I changed that!

**Comment by ganwell** [comment]
@sliverc  Sorry, the nvim was sitting there with :w (no enter)

**Pull request merged**